dnfdaemon-client/commands/command.cpp

```diff
@@ -39,11 +42,43 @@ void TransactionCommand::run_transaction(Context & ctx) {
     // resolve the transaction
     options["allow_erasing"] = ctx.allow_erasing.get_value();
     std::vector<dnfdaemon::DbusTransactionItem> transaction;
+    dnfdaemon::KeyValueMap dbus_goal_resolve_results;
     ctx.session_proxy->callMethod("resolve")
         .onInterface(dnfdaemon::INTERFACE_GOAL)
         .withTimeout(static_cast<uint64_t>(-1))
         .withArguments(options)
-        .storeResultsTo(transaction);
+        .storeResultsTo(transaction, dbus_goal_resolve_results);
+
+    if (static_cast<libdnf::GoalProblem>(
+            key_value_map_get<uint32_t>(dbus_goal_resolve_results, "transaction_problems")
+            ) != libdnf::GoalProblem::NO_PROBLEM) {
```

Comment:
I think the problems should be printed out regardless `transaction_problems != libdnf::GoalProblem::NO_PROBLEM`.

There are cases when solver returns `NO_PROBLEM`, you got a transaction but simultaneously some non-fatal solver problems are returned. For example when you try to upgrade your system with some excludes, you might get something like this:

```
$ sudo dnf upgrade --exclude=kernel-core
Dependencies resolved.

 Problem 1: package kernel-modules-extra-5.13.13-200.fc34.x86_64 requires kernel-uname-r = 5.13.13-200.fc34.x86_64, but none of the providers can be installed
  - cannot install the best update candidate for package kernel-modules-extra-5.13.5-200.fc34.x86_64
  - package kernel-core-5.13.13-200.fc34.x86_64 is filtered out by exclude filtering
 Problem 2: package kernel-modules-extra-5.13.13-200.fc34.x86_64 requires kernel-uname-r = 5.13.13-200.fc34.x86_64, but none of the providers can be installed
  - cannot install the best update candidate for package kernel-modules-extra-5.13.8-200.fc34.x86_64
  - package kernel-core-5.13.13-200.fc34.x86_64 is filtered out by exclude filtering
  
....

=========================================================================================
 Package         Arch   Version            Repository                               Size
=========================================================================================
Upgrading:
 buildah         x86_64 1.22.3-1.fc34      updates                                 7.5 M
.....
Skipping packages with broken dependencies:
 kernel          x86_64 5.13.13-200.fc34   updates                                  72 k
 kernel-modules  x86_64 5.13.13-200.fc34   updates                                  32 M
 kernel-modules-extra
                 x86_64 5.13.13-200.fc34   updates                                 2.0 M

Transaction Summary
=========================================================================================
Upgrade  15 Packages
Skip      3 Packages

Total download size: 21 M
Is this ok [y/N]:
```

The other case might be an attempt to upgrade a non-installed package. The `resolve()` would return empty transaction and `NO_PROBLEM`. But there still is this information `Packages for argument 'non-installed-package' available, but not installed.` to display to the user.

Reply:
the `if` is just checking nothing to be fair. so I removed it and now it's reporting all the problems